src/components/ImageGrid/ImageGridComponent.tsx

```diff
@@ -98,29 +73,147 @@ const ImageGridComponent = ({
     } else {
       setRowCol({ col: 1, row: 1 });
     }
-  }, [numberOfImgs, images]);
-
-  // to avoid bug undefined url when user change number of images
-  useEffect(() => {
-    for (let i = 0; i < numberOfImgs; i++) {
-      images[i] = exampleImages[i];
-    }
   }, [numberOfImgs]);
+  // close Carousel when user press esc
+  useEffect(() => {
+    const keydownHandler = (evnt: KeyboardEvent) => {
+      evnt.key === "Escape" && setShowCarousel(false);
+    };
+    document.addEventListener("keydown", keydownHandler);
+    return () => document.removeEventListener("keydown", keydownHandler);
+  });
 
+  const useCarousel = (index: number) => {
+    showModal && setShowCarousel(true);
```

Comment:
why do you need both showModal & showCarousel?

Reply:
bcs I did an option showModal in the component, to let users choose whether they wanna show the carousel or not in the attribute of the component. 
by default, this attribute is true 
Hm... but maybe, this option is not necessary. I will remove the showModal attribute.